openlibrary/core/search.py

```diff
@@ -0,0 +1,179 @@
+        'identifierSorter', 'imagecount', 'indexdate',
+        'item_size', 'languageSorter', 'licenseurl', 'mediatype',
+        'mediatypeSorter', 'month', 'nav_order', 'num_reviews',
+        'programSorter', 'publicdate', 'reviewdate', 'stars',
+        'titleSorter', 'titleSorterRaw', 'week',  'year',
+        'loans__status__last_loan_date'
+    ]
+    if sorts:
+        # If it's a string, split and turn to list
+        if isinstance(sorts, six.string_types):
+            sorts = sorts.split(',')
+        if isinstance(sorts, list):
+            # compare against field with +asc/desc suffix rm'd
+            # e.g: date, not date+asc or date+desc
+            return [sort.replace('+', ' ') for sort in sorts
+                    if sort.split('+')[0] in VALID_SORTS]
+    return ['']
+
+def _expand_api_query(query):
+    """Expands short / more easily cacheable preset queries
+    and fixes query to ensure only text archive.org items
+    are returned having openlibrary_work
+    """
+    # Expand preset queries
+    if query in PRESET_QUERIES:
+        query = PRESET_QUERIES[query]
+
+    q = 'mediatype:texts AND !noindex:* AND openlibrary_work:(*)'
+    # Add availability if none present (e.g. borrowable only)
+    if AVAILABILITY_STATUS not in query:
+        q += ' AND %s:AVAILABLE' % AVAILABILITY_STATUS
+    return q if not query else q + " AND " + query
+
+def _clean_params(q='', sorts='', page=1, limit=MAX_IA_RESULTS):
+    params = {
+        'q': q,
+        'page': page,
+        'rows': min(limit, MAX_IA_RESULTS),
+        'sort[]': sorts,  # broken for encoding of + -> %2B
+        'fl[]': RETURN_FIELDS,
+        'output': 'json'
+    }
+    return params
+
+def _compose_advancedsearch_url(**params):
+    return SEARCH_URL + '?' + urllib.urlencode(params, doseq=True)
+
+def _compose_browsable_url(query, sorts=None):
+    """If the client wants to explore this query in the browser, it will
+    have to be converted to user a human readable version of the
+    advancedsearch API
+    """
+    _sort = sorts[0] if sorts else ''
+    if ' desc' in _sort:
+        _sort = '-' + _sort.split(' ')[0]
+    elif ' asc' in _sort:
+        _sort = _sort.split(' ')[0]
+    return '%s?query=%s&sort=%s' % (BROWSE_URL, query, _sort)
+
+def _request(url):
+    """Hits archive.org advancedsearch.php API, returns matching items
+    """
+    try:
+        request = urllib2.Request(url)
+        # Internet Archive Elastic Search (which powers some of our
+        # carousel queries) needs Open Library to forward user IPs so
+        # we can attribute requests to end-users
+        client_ip = web.ctx.env.get('HTTP_X_FORWARDED_FOR', 'ol-internal')
+        request.add_header('x-client-id', client_ip)
+
+        response = urllib2.urlopen(
+            request, timeout=config_http_request_timeout).read()
+        return simplejson.loads(response).get('response', {})
+    except Exception as e:
+        return []
```

Comment:
Should this be `{}`? or maybe `{"status": "error"}`?
```suggestion
        return {}
```

Reply:
some sort of error would be nice, this buries errors and will make noticing problems and debugging hard.